library/src/main/scala/za/co/absa/springdocopenapiscala/OpenAPIModelRegistration.scala

```diff
@@ -144,13 +144,57 @@ class OpenAPIModelRegistration(
     s.isTerm && s.asTerm.isVal && s.typeSignature <:< typeOf[Enumeration#Value]
 
   private def handleSealedType(tpe: Type): Schema[_] = {
+
+    def addDiscriminatorPropertyToChildren(
+      currentSchema: Schema[_],
+      discriminatorPropertyName: String,
+      addOnlyToDirectChildren: Boolean,
+      discriminatorValue: Option[String] = None
+    ): Unit = {
+      val children = currentSchema.getOneOf.asScala
+      children.foreach { s =>
+        val ref = s.get$ref
+        val name = extractSchemaNameFromRef(ref)
+        val actualSchema = components.getSchemas.get(name)
+        if (actualSchema.getType == "object") {
+          val constEnumSchema = createConstEnumSchema(discriminatorValue.getOrElse(name))
+          actualSchema.addProperty(discriminatorPropertyName, constEnumSchema)
+          actualSchema.addRequiredItem(discriminatorPropertyName)
+        } else if (!addOnlyToDirectChildren && Option(actualSchema.getOneOf).map(!_.isEmpty).getOrElse(false)) {
```

Comment:
perhaps add an explanatory comment for a non-trivial condition?

Reply:
Added

Reply:
Much better, thanks!